feat: Add Web Vitals documentation (#10401)
PostHog/posthog.com: contents/docs/web-analytics/web-vitals.mdx

### INP - Interaction to Next Paint

Interaction to Next Paint (INP) measures the responsiveness of your site by measuring the time it takes to respond to user interactions like clicks or taps. It's important because it tells you how quickly your site reacts to user input, which is a key factor in user experience.

A good INP score is 100ms or less. Poor INP scores can be caused by slow JavaScript execution, blocking the main thread, or excessive layout thrashing. To improve INP, optimize JavaScript performance, reduce unnecessary layout changes, and ensure smooth animations.

## Web vitals dashboard

You can access web vitals in the sidebar by clicking on the dropdown to the right of web analytics.

If you're inside web analytics, you can also toggle to the web vitals view by clicking the toggle in the top right.

> > TODO: ADD IMAGE

Author: Missing this, working on it, needs to merge https://github.com/PostHog/posthog/pull/27691 first

## Web vitals dashboard

You can access web vitals in PostHog's sidebar by clicking on the dropdown to the right of web analytics.

Alternatively, if you're inside web analytics, you can also toggle to the web vitals view by clicking the toggle in the top left.

The dashboard provides you with a summary of your web vitals and allows you to drill down into the data.

<ProductScreenshot
  imageLight="https://res.cloudinary.com/dmukukwp6/image/upload/Captura_de_tela_2025_01_21_104838_a4f42373ca.png"
  imageDark="https://res.cloudinary.com/dmukukwp6/image/upload/Captura_de_tela_2025_01_21_104948_29c77b1863.png"
  alt="Web vitals dashboard" 
  classes="rounded"
/>

In this dashboard, you can see how that data changed over time for the 4 different metrics we keep track of, allowing you to specify a time period, and also add the same filters you can already set in Web Analytics.

At the bottom of the dashboard, you can see a table with a breakdown of the pages with good, needs improvement, and poor scores. You should focus on optimizing the pages with poor scores.

### What are the thresholds for good Web Vitals?

> Web Vitals can be categorized in 3 categories: **Good**, **Needs improvement**, and **Poor**.

Reviewer: ```suggestion
Web vitals can be categorized in 3 categories: **Good**, **Needs improvement**, and **Poor**.
```

## How to enable web vitals autocapture

Go to your [Autocapture settings](https://app.posthog.com/settings/project-autocapture#web-vitals-autocapture). Then, in the **Web vitals autocapture** section, click **Enable**.

> **Note:** Web vitals autocapture is separate from our regular [autocapture](/docs/product-analytics/autocapture) feature. Web vitals can be captured regardless of whether autocapture is enabled.

> **Note:** You'll need to be using SDK v1.141.2+ to use web vitals autocapture.

<ProductScreenshot
  imageLight="https://res.cloudinary.com/dmukukwp6/image/upload/v1726058265/posthog.com/contents/Screenshot_2024-09-11_at_1.36.27_PM.png"
  imageDark="https://res.cloudinary.com/dmukukwp6/image/upload/v1726058265/posthog.com/contents/Screenshot_2024-09-11_at_1.36.34_PM.png"
  alt="How to enable web vitals autocapture" 
  classes="rounded"
/>

Once enabled, web vitals are captured under the `$web_vitals` event name. These events include properties for each of the metrics like `$web_vitals_FCP_value` and `$web_vitals_FCP_event`. 

You can choose which metrics to capture by using the toggles.

### Configuring web vitals autocapture

> You can provide a client side config to tune web vitals autocapture. These are set on [the `capture_performance` key](https://github.com/PostHog/posthog-js/blob/main/src/types.ts#L209).

Reviewer: It looks like the line number changed for the `capture_performance` key. I wonder if we should maybe just point to the file without the line number.

```suggestion
You can provide a client side config to tune web vitals autocapture. These are set on [the `capture_performance` key](https://github.com/PostHog/posthog-js/blob/main/src/types.ts) in the `PostHogConfig` interface.
```

| `web_vitals_allowed_metrics`<br/>default: `['LCP', 'CLS', 'FCP', 'INP' ]` | By default all 4 metrics are captured. You can set this config to restrict which metrics are captured. e.g. `['CLS', 'FCP']` to only capture those two metrics. This does not override whether the capture is enabled |
| `web_vitals_delayed_flush_ms`<br/>default: `5000`                         | We delay flushing web vitals metrics to reduce the number of events we send. This is the maximum time we will wait before sending the metrics.                                                                        |

## When and how are events captured?

The SDK attempts to send as many of the available metrics as possible in each `$web_vitals` event. By default, we wait up to 5 seconds since the page load or last `$web_vitals` event to try and catch all four metrics in one event. See the section on [configuring web vitals](#configuring-web-vitals-autocapture) capture to change this behaviour.

Metrics are made available when the [web vitals library](https://github.com/GoogleChrome/web-vitals) triggers them:

> - FCP and LCP are relative to a page being rendered and are made available relatively soon after loading a page. 

Reviewer: ```suggestion
- FCP and LCP are relative to a page being rendered and are made available soon after loading a page. 
```

- INP and CLS are measured across the lifecycle of a page and don't have a fixed time they become available.

> You can look at our Web vitals capturing code in here: [web-vitals/index.ts](https://github.com/PostHog/posthog-js/blob/main/src/extensions/web-vitals/index.ts).

Reviewer: ```suggestion
You can look at our Web vitals capturing code in [web-vitals/index.ts](https://github.com/PostHog/posthog-js/blob/main/src/extensions/web-vitals/index.ts).
```

## Pricing

> Web vitals (`$web_vitals`) events are charged as regular events, first 1M are free. Check our [pricing page](/pricing) for more information, we have a calculator there too! 

Reviewer: ```suggestion
Web vitals (`$web_vitals`) events are charged as regular events, and the first 1M are free. Check our [pricing page](/pricing) for more information, and we have a calculator there too!
```
